### DISCUSSION TITLE
Building model for 4-band data?

### DISCUSSION THREAD

**Discussion by NoaMills-USDA** [azavea/raster-vision#1472] [Q&A] [answered]
I'm trying to train rastervision to do semantic segmentation on 4 band imagery. I obviously can't use resnet50 for this task, since resnet50 is built for 3 band imagery. How can I create a custom neural network model to use in resnet50's place?

**Comment by AdeelH** [comment, marked as answer]
Raster Vision is quite capable of handling arbitrary number of bands. See the `multiband` argument is [`isprs_potsdam.py`](https://github.com/azavea/raster-vision/blob/master/rastervision_pytorch_backend/rastervision/pytorch_backend/examples/semantic_segmentation/isprs_potsdam.py) for an example. 

To use a custom model architecture, see the `external_model` argument in the same file.

**Reply by NoaMills-USDA** [reply]
Thanks for the info! I was able to get my model running through the train and bundle stages but I'm getting a runtime error at the prediction stage. I'll start a new discussion thread to follow up.

**Reply by NoaMills-USDA** [reply]
Also, for documentation, I am running rastervision in a singularity container on a compute cluster, and in order to get the model to train, I had to chmod +w the files in pytorch-fpn/.git/objects/pack because I was getting permission denied errors. Would love to know what these files are for.

Further, I was getting another permission denied error at the bundle stage, and found that the model was trying to write to the /local/scratch folder, which is in the singularity container and by default read-only. I was able to fix this by bind mounting to the /local folder in the container.
